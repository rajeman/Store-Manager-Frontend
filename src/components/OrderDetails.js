import React from 'react';
import { connect } from 'react-redux';
import { fetchSale } from '../actions/records';
import Record from './Record';
import OrderDetail from './OrderDetail';

class OrderDetails extends React.Component {
  componentDidMount() {
    this.props.dispatch(fetchSale(this.props.id));
  }
  render() {
    const { sale } = this.props.sales;
    return (
      <div className="container">
        <div className="wrapper">
          <h2>Order Summary</h2>
          <table id="orders-table">
            <tr>
              <th>Attendant ID</th>
              <th>Attendant Name</th>
              <th>Order Id</th>
              <th>Time Created</th>
              <th>Total Item</th>
              <th>Order Price</th>
            </tr>
            {sale[0] && <Record sale={sale[0]} key={sale.order_id} />}
          </table>
          <h2>Details</h2>
          <table id="details-table" >
            <tr>
              <th>Product ID</th>
              <th>Product Name</th>
              <th>Quantity</th>
              <th>Price per Product</th>
              <th>Total Price</th>
            </tr>
            {sale.map((detail) =>
              (
                <OrderDetail detail={detail} />
              )
            )}
          </table>
        </div>
      </div>

    );
  }
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps)(OrderDetails);



