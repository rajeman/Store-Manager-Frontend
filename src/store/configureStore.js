import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import productReducer from '../reducers/products';
import authReducer from '../reducers/auth';
import saleReducer from '../reducers/records';
import navigationReducer from '../reducers/navigation';
import attendantReducer from '../reducers/attendants';
import cartReducer from '../reducers/cart';

export default () => createStore(
  combineReducers({
    products: productReducer,
    auth: authReducer,
    navigation: navigationReducer,
    sales: saleReducer,
    attendants: attendantReducer,
    cart: cartReducer,
  }), applyMiddleware(thunk),
);
