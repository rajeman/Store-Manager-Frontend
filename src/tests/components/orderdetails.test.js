import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import axios from 'axios';
import OrderDetails from '../../components/OrderDetails';

jest.mock('axios');

describe('ORDER DETAILS TEST SUITE', () => {
  it('should render the OrderDetails component', () => {
    const response = {
      data:
      {
        message: 'successfully fetched user info',
      }
    };
    axios.get.mockResolvedValue(response);
    const initialState = {
      sales: {
          sale: [{order_id: 25}]
      }
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    const component = mount(
      <Provider store={store}>
        <OrderDetails />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
