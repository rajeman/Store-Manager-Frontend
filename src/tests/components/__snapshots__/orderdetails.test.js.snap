// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ORDER DETAILS TEST SUITE should render the OrderDetails component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(OrderDetails)>
    <OrderDetails
      dispatch={[Function]}
      sales={
        Object {
          "sale": Array [
            Object {
              "order_id": 25,
            },
          ],
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="wrapper"
        >
          <h2>
            Order Summary
          </h2>
          <table
            id="orders-table"
          >
            <tr>
              <th>
                Attendant ID
              </th>
              <th>
                Attendant Name
              </th>
              <th>
                Order Id
              </th>
              <th>
                Time Created
              </th>
              <th>
                Total Item
              </th>
              <th>
                Order Price
              </th>
            </tr>
            <Record
              sale={
                Object {
                  "order_id": 25,
                }
              }
            >
              <tr
                onClick={[Function]}
              >
                <td />
                <td />
                <td>
                  25
                </td>
                <td>
                  undefined-NaN-NaN NaN:aN
                </td>
                <td />
                <td />
              </tr>
            </Record>
          </table>
          <h2>
            Details
          </h2>
          <table
            id="details-table"
          >
            <tr>
              <th>
                Product ID
              </th>
              <th>
                Product Name
              </th>
              <th>
                Quantity
              </th>
              <th>
                Price per Product
              </th>
              <th>
                Total Price
              </th>
            </tr>
            <OrderDetail
              detail={
                Object {
                  "order_id": 25,
                }
              }
            >
              <tr>
                <td />
                <td />
                <td />
                <td />
                <td />
              </tr>
            </OrderDetail>
          </table>
        </div>
      </div>
    </OrderDetails>
  </Connect(OrderDetails)>
</Provider>
`;
